src/installers/GDWeaveInstaller.ts

```diff
@@ -0,0 +1,138 @@
+
+        for (const fileOrFolder of toCopy) {
+            await FileUtils.copyFileOrFolder(
+                path.join(root, fileOrFolder),
+                profile.joinToProfilePath(fileOrFolder)
+            );
+        }
+    }
+
+    async uninstall(args: InstallArgs): Promise<void> {
+        const { profile } = args;
+
+        try {
+            // ReturnOfModding deletes all files in the root and ignores directories.
+            // We can't do that because GDWeave/core exists, but blanket deleting GDWeave would delete all mods in the process.
+            // This would probably be handled better by a state file, but /shrug.
+
+            const toDelete = [
+                profile.joinToProfilePath('winmm.dll'),
+                profile.joinToProfilePath('GDWeave/core'),
+            ];
+            for (const fileOrFolder of toDelete) {
+                if (!(await FsProvider.instance.exists(fileOrFolder))) {
+                    continue;
+                }
+
+                if ((await FsProvider.instance.lstat(fileOrFolder)).isFile()) {
+                    await FsProvider.instance.unlink(fileOrFolder);
+                } else {
+                    await FileUtils.recursiveRemoveDirectoryIfExists(
+                        fileOrFolder
+                    );
+                }
+            }
+        } catch (e) {
+            const name = 'Failed to delete GDWeave files from profile root';
+            const solution = 'Is the game still running?';
+            throw FileWriteError.fromThrownValue(e, name, solution);
+        }
+    }
+}
+
+async function searchForManifest(
+    dir: string,
+    topLevel: boolean = true
+): Promise<string | null> {
+    // Ignore `manifest.json` at the top level - that's Thunderstore's
+    if (
+        !topLevel &&
+        (await FsProvider.instance.exists(path.join(dir, 'manifest.json')))
+    ) {
+        return dir;
+    }
+
+    for (const item of await FsProvider.instance.readdir(dir)) {
+        if (
+            (await FsProvider.instance.stat(path.join(dir, item))).isDirectory()
+        ) {
+            const result = await searchForManifest(path.join(dir, item), false);
+            if (result) return result;
+        }
+    }
+
+    return null;
+}
+
+export class GDWeavePluginInstaller implements PackageInstaller {
+    async install(args: InstallArgs) {
+        const { mod, packagePath, profile } = args;
+
+        // Packaging is all over the place. Find a folder with a manifest.json
+        // not at the top level (because the top level is Thunderstore's packaging)
+        const modFolder = await searchForManifest(packagePath);
+        if (!modFolder) {
+            throw new Error('Could not find mod folder');
+        }
+
+        const root = profile.joinToProfilePath(
+            'GDWeave',
+            'mods',
+            mod.getName()
+        );
+        await FileUtils.ensureDirectory(root);
```

Comment:
I think `FsProvider.instance.copyFolder(modFolder, root);` might be sufficient to replace lines 111-119?